src/dumbo/src/ns.rs

```diff
@@ -139,35 +139,38 @@ impl MmdsNetworkStack {
         // checksum computation from the guest driver to some other entity. Clear up this entire
         // context at some point!
         if let Ok(ip) = IPv4Packet::from_bytes(eth.payload(), false) {
-            if ip.destination_address() == self.ipv4_addr {
-                if ip.protocol() == PROTOCOL_TCP {
-                    self.remote_mac_addr = eth.src_mac();
-                    match self.tcp_handler.receive_packet(&ip) {
-                        Ok(event) => {
-                            METRICS.mmds.rx_count.inc();
-                            match event {
-                                RecvEvent::NewConnectionSuccessful => {
-                                    METRICS.mmds.connections_created.inc()
-                                }
-                                RecvEvent::NewConnectionReplacing => {
-                                    METRICS.mmds.connections_created.inc();
-                                    METRICS.mmds.connections_destroyed.inc();
-                                }
-                                RecvEvent::EndpointDone => {
-                                    METRICS.mmds.connections_destroyed.inc();
-                                }
-                                _ => (),
+            if ip.protocol() == PROTOCOL_TCP {
+                // Note-1: `remote_mac_address` is actually the network device mac address, on which
+                // this TCP segment came from.
```

Comment:
Confusing phrasing, suggest
```suggestion
                // Note-1: `remote_mac_address` is actually the network device mac address,
                // where this TCP segment came from.
```